src/test/java/org/springframework/integration/kafka/support/KafkaConsumerContextIntegrationTests.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.kafka.support;
+
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.UUID;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.springframework.integration.kafka.outbound.KafkaProducerMessageHandler;
+import org.springframework.integration.kafka.rule.KafkaEmbedded;
+import org.springframework.integration.kafka.rule.KafkaRule;
+import org.springframework.integration.kafka.serializer.common.StringDecoder;
+import org.springframework.integration.kafka.serializer.common.StringEncoder;
+import org.springframework.integration.kafka.util.EncoderAdaptingSerializer;
+import org.springframework.integration.kafka.util.TopicUtils;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+
+import kafka.admin.AdminUtils;
+import kafka.common.TopicExistsException;
+import kafka.consumer.ConsumerConfig;
+import kafka.serializer.Encoder;
+
+/**
+ * @author Gary Russell
+ * @since 1.3
+ */
+public class KafkaConsumerContextIntegrationTests {
+
+	private static final String TOPIC = "springintegrationtestInbound";
+
+	@Rule
+	public KafkaRule kafkaRule = new KafkaEmbedded(1);
+
+	@After
+	public void tearDown() {
+		try {
+			AdminUtils.deleteTopic(kafkaRule.getZkClient(), TOPIC);
+		}
+		catch (Exception e) {
+		}
+	}
```

Comment:
I think this must be `@Before` as well. In case of crash for previous cycle we should be sure that the new one will be started on a fresh environment.